test/unit/core/Object3D.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @author simonThiele / https://github.com/simonThiele
+ */
+
+module( "Object3D" );
+
+test( "getWorldScale", function() {
+  var obj = new THREE.Object3D();
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(0, 0, 0);
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(-1, -12.4324, 32456323);
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+});
+
+test( "getWorldScale with hierarchy", function() {
+  var obj = new THREE.Object3D();
+  var parent1 = new THREE.Object3D();
+  var parent2 = new THREE.Object3D();
+
+  parent2.add(parent1);
+  parent1.add(obj);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(0, 0, 0);
+  parent2.scale.set(-1, -12.4324, 32456323);
+  parent1.scale.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.scale.set(1, 0, -1);
+  parent2.scale.set(-1, -12.4324, 32456323);
+  parent1.scale.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+
+  obj.position.set(1, 0, -1);
+  parent2.position.set(-1, -12.4324, 32456323);
+  parent1.position.set(10, 20, -234);
+
+  checkIfDecomposeAndGetWorldScaleAreEqual(obj);
+});
+
+function checkIfDecomposeAndGetWorldScaleAreEqual(obj) {
+  var result = new THREE.Vector3();
+
+  // this is the old way to get scale: decompose the complete matrix and return scale part
```

Comment:
Maybe you could re-phrase these comments, removing the reference to old and new ways, which are quite specific to this PR.